### PR TITLE
New book branch

### PR DESCRIPTION
**Pull request by jheart-vic**
- Created the book class and it's method
- Created the label class ad it's method
- Created an Addbook class
- Created an Addlabel class
- Created Json files to preserve data
- Created a Book and Label store modules

### CODE REVIEW

File: .github/workflows/book.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+require './item'
+
+# the book class
+class Book < Item
+  attr_accessor :publisher
+
+  def initialize(id, publisher, cover_state)
+    super()
```
Comment (sabetech) [inline .github/workflows/book.rb:8]: Super should pass the arguments needed for an item to be created
Eg: `super(publish_date, id, archived: archived)`

File: .github/workflows/book.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  def initialize(id, publisher, cover_state)
+    super()
+    @id = id || Random.rand(1...1000)
+    @publisher = publisher
+    @cover_state = cover_state
+  end
+
+  private
+
+  def can_be_archieved?
```
Comment (sabetech) [inline .github/workflows/book.rb:16]: method name should be `can_be_archived?`

File: .github/workflows/book.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    return true if super || cover_state == 'bad'
+
+    false
+  end
```
Comment (bobb-Rob) [inline .github/workflows/book.rb:20]: Great work on this project.

- [ ] But I think that the method code can be shorter than what you have here. See below my suggestion which does the same thing. 

Suggesting this 👇 
```
def can_be_archieved?
    super || cover_state == 'bad'   
 end
```

File: label.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    super()
+    @id = id || Random.rand(1...1000)
+    @title = title
+    @color = color
+    @items = []
+  end
+
+  def add_item(item)
+    @items << item
+    Item.label = self
```
Comment (bobb-Rob) [inline label.rb]: This line should reference the passed-in item and not the Item class

**Review by bobb-Rob — APPROVED** [review]
## STATUS: APPROVED ✔️ 🍾 🍾 🍾 🥳 🥳 🥳 

Hi , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
✅ No Linter error
✅ Correct use of GitHub flow
✅ Very descriptive Pull Request(PR)
✅ Neat Commit messages

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**